### PR TITLE
feat: 🎸 drag file to editor trigger recompile

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
add two feature :
1. drag file to editor now will trigger reparse the wasm to wat
2. cache the `features` user checked, so user no need to reselect the features every open.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
b0fos80OvH

**Comment by IWANABETHATGUY** [comment]
@binji could you please help for review?

File: docs/demo/wasm2wat/demo.js [hunk 1]
```diff
@@ -14,7 +14,17 @@
  * limitations under the License.
  */
 
-var features = {};
+
+var features = getLocalStorageFeatures()
+
+function getLocalStorageFeatures() {
+  try {
+    return JSON.parse(localStorage && localStorage.getItem("features")) || { mutable_globals: true };
```
Comment (binji) [inline docs/demo/wasm2wat/demo.js:22]: Can you factor out the `{mutable_globals:true}` (here and below) into a separate `defaultFeatures`?

File: docs/demo/wasm2wat/demo.js [hunk 2]
```diff
@@ -40,19 +50,28 @@ var generateNamesEl = document.getElementById('generateNames');
...
-
 var options = {mode: 'wast', lineNumbers: true};
 var editor = CodeMirror.fromTextArea(editorEl, options);
 
-var fileBuffer = null;
+var editorContainer = document.querySelector('.CodeMirror.cm-s-default')
 
+editorContainer.ondrop = function(e) {
+  e.preventDefault();
+  let file = e.dataTransfer.files[0]
```
Comment (binji) [inline docs/demo/wasm2wat/demo.js:60]: nit: use semicolons (here and elsewhere)

File: docs/demo/wasm2wat/demo.js [hunk 3]
```diff
@@ -40,19 +50,28 @@ var generateNamesEl = document.getElementById('generateNames');
...
+var fileBuffer = null;
 for (var feature of FEATURES) {
   var featureEl = document.getElementById(feature);
-  features[feature] = featureEl.checked;
+  featureEl.checked = !!features[feature]
   featureEl.addEventListener('change', event => {
     var feature = event.target.id;
     features[feature] = event.target.checked;
     compile(fileBuffer);
+    localStorage && localStorage.setItem('features', JSON.stringify(features))
```
Comment (binji) [inline docs/demo/wasm2wat/demo.js:74]: nit: use `if` instead of `&&`

**Review by binji — COMMENTED** [review]
Thanks for adding this (and the cool gif)! Sorry for the delay in review. Just a few small comments, otherwise looks good!

**Comment by IWANABETHATGUY** [comment]
I have fixed these problem.

**Review by binji — APPROVED** [review]

**Pull request merged**